gobblin-metastore/src/main/java/org/apache/gobblin/metastore/DatasetStoreDataset.java

```diff
@@ -36,7 +38,8 @@
 
   @Override
   public String datasetURN() {
-    return this.key.getStoreName() + ":::" + this.key.getSanitizedDatasetUrn();
+    return this.key.getStoreName() +
+        (StringUtils.isEmpty(this.key.getSanitizedDatasetUrn()) ? "" : ":::" + this.key.getSanitizedDatasetUrn());
```

Comment:
Nit: Guava's Strings.isNullOrEmpty() is more readable. I did not realize until I saw the Javadoc for StringUtils.isEmpty that it returns true if the string is null.